common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -757,14 +757,20 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
                 "User [%s] performed SSO through [%s] who synchronizes profile data, and the "
                 "following fields were changed: %s", user.username, current_provider.name, changed.keys(),
             )
-            try:
-                # Save changes to user and user.profile models.
-                strategy.storage.user.changed(user)
-                user.profile.save()
-            except IntegrityError as exc:
-                logger.warning('User [%s] tried to synchronize profile data through [%s] '
-                               'but there was a conflict: %s', user.username, current_provider.name, str(exc))
-                return
+
+            # For any conflicts that would raise an integrity error, undo the change.
+            integrity_conflict_fields = {'email': user.email, 'username': user.username}
+            for field, value in integrity_conflict_fields.items():
+                if User.objects.filter(**{field: value}).exists() and changed.get(field):
```

Comment:
Shouldn't we just do this check above where we are originally setting the values on the User model fields?

Reply:
Good idea. :+1: